crates/ef-testing/src/evm_sequencer/account/v1.rs

```diff
@@ -1,56 +1,28 @@
 use blockifier::abi::abi_utils::get_storage_var_address;
-use reth_primitives::{Address, Bytes, U256};
+use blockifier::abi::sierra_types::next_storage_key;
+use reth_primitives::{keccak256, Address, Bytes, KECCAK_EMPTY, U256};
+use revm_interpreter::analysis::to_analysed;
+use revm_primitives::Bytecode;
 use starknet::core::types::Felt;
 use starknet_api::{core::Nonce, state::StorageKey, StarknetApiError};
 
 use super::KakarotAccount;
 use super::{inner_byte_array_pointer, pack_byte_array_to_starkfelt_array};
 use crate::evm_sequencer::constants::storage_variables::{
-    ACCOUNT_EVM_ADDRESS, ACCOUNT_IS_INITIALIZED, ACCOUNT_NONCE, ACCOUNT_STORAGE,
+    ACCOUNT_BYTECODE_LEN, ACCOUNT_CODE_HASH, ACCOUNT_EVM_ADDRESS, ACCOUNT_IS_INITIALIZED,
+    ACCOUNT_NONCE, ACCOUNT_STORAGE, ACCOUNT_VALID_JUMPDESTS,
 };
 use crate::evm_sequencer::{
     constants::storage_variables::ACCOUNT_BYTECODE, types::felt::FeltSequencer, utils::split_u256,
 };
-use crate::{
-    evm_sequencer::evm_state::v1::{compute_storage_base_address, offset_storage_key},
-    starknet_storage,
-};
-
-/// The layout of a `ByteArray` in storage is as follows:
-/// * Only the length in bytes is stored in the original address where the byte array is logically
-///   stored.
-/// * The actual data is stored in chunks of 256 `bytes31`s in another place in storage
-///   determined by the hash of:
-///   - The address storing the length of the array.
-///   - The chunk index.
-///   - The short string `ByteArray`.
-fn prepare_bytearray_storage(code: &Bytes) -> Vec<(StorageKey, Felt)> {
-    let bytecode_base_address = get_storage_var_address(ACCOUNT_BYTECODE, &[]);
-    let mut bytearray = vec![(bytecode_base_address, Felt::from(code.len()))];
-
-    let bytecode_storage: Vec<_> = pack_byte_array_to_starkfelt_array(code)
-        .enumerate()
-        .map(|(index, b)| {
-            let offset = index % 256;
-            let index = index / 256;
-            let key = inner_byte_array_pointer(*bytecode_base_address.0.key(), index.into());
-            (
-                offset_storage_key(key.try_into().unwrap(), offset as i64),
-                b,
-            )
-        })
-        .collect();
-    bytearray.extend(bytecode_storage);
-
-    bytearray
-}
+use crate::starknet_storage;
 
 impl KakarotAccount {
     pub fn new(
```

Comment:
If not mistaken this now completely identical to v0 
Should this be refactored to only have one file and avoid v0 / v1 ?

Reply:
eventually yes but before deleting everything I want to make sure I won't have other changes to make later

Reply:
Ok good for me then